### PR TITLE
Bugfix/artv3/align

### PR DESCRIPTION
**Pull request by artv3**
Fixed bug in align. Created common interface for GCC/Clang/Intel. GCC/Clang returns void * while Intel returns nothing.

### CODE REVIEW

File: include/RAJA/config.hpp.in [hunk 1]
```diff
@@ -277,6 +277,18 @@ const int DATA_ALIGN = @RAJA_DATA_ALIGN@;
 #cmakedefine RAJA_HAVE_ALIGNED_ALLOC
 #cmakedefine RAJA_HAVE_MM_MALLOC
 
+template<typename T>
+RAJA_INLINE
+T * align_hint(T * x)
```
Comment (davidbeckingsale) [inline include/RAJA/config.hpp.in]: Maybe I'm confused, but I don't know where this is being used?

Comment (artv3) [reply include/RAJA/config.hpp.in]: Currently just in a unit test. But the issue is that when adding alignment hints, GCC and clang will return a void * while Intel will return nothing. This allows a common interface for alignment memory hints.

Comment (artv3) [reply include/RAJA/config.hpp.in]: In the scenario that CUDA is enabled, I opted for removing compiler alignment hints. I am not sure what the best practice would be here.

**Comment by artv3** [comment]
I updated this branch with more documentation on the common framework for compiler alignment hints.

**Pull request merged**